src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Application {
+    public static void main(String[] args) {
+        int totalNumberOfFigures = 11;
+
+        Figure [] result = new Figure[totalNumberOfFigures];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < totalNumberOfFigures / 2; i++) {

```

Comment:
The code doesn't account for cases where totalNumberOfFigures is an odd number. This may lead to off-by-one errors.

Reply:
Why do you think so? 
In past version there was mechanizm if-else but you recommended to reduce it and I think you had right and now it's steel working correct.